src/readout/mqtt.go

```diff
@@ -15,7 +15,7 @@ func newMQTTClient(c *smartpi.Config) (mqttclient MQTT.Client) {
 	log.Debugf("Connecting to MQTT broker at %s", (c.MQTTbroker + ":" + c.MQTTbrokerport))
 	//create a MQTTClientOptions struct setting the broker address, clientid, user and password
 	opts := MQTT.NewClientOptions().AddBroker("tcp://" + c.MQTTbroker + ":" + c.MQTTbrokerport)
-	opts.SetClientID("SmartPi")
+	opts.SetClientID(c.Serial)
```

Comment:
Good point! At the moment the default serial already contains a smartpi prefix (`smartpi170512345`), but of course this string is not fixed.

Instead of `"SmartPi-" + c.Serial` we might use `"SmartPi-" + c.Name` as client identifier.

Reply:
`"SmartPi-" + c.Name` sounds better to me.